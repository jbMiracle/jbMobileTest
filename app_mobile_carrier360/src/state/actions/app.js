import { App as AppActionTypes } from '../types';

export const genesis = () => ({
  type: AppActionTypes.Genesis,
});

export const forgotPassword = email => ({
  type: AppActionTypes.ForgotPassword,
  payload: email,
});

export const forgotPasswordServiceDone = (success, errorMessage) => ({
  type: AppActionTypes.ForgotPasswordDone,
  payload: {
    success,
    errorMessage,
  },
});
