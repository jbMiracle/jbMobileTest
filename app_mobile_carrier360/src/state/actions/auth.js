import { Auth as AuthActionTypes } from '../types';

export const userAuthenticated = (result = true) => ({
  type: AuthActionTypes.UserAuthenticated,
  payload: result,
});

export const queryBiometricSupport = () => ({
  type: AuthActionTypes.QueryBiometricSupport,
});

export const biometricSupportDetermined = supportResult => ({
  type: AuthActionTypes.BiometricSupportDetermined,
  payload: supportResult,
});

export const setupBiometricLogin = () => ({
  type: AuthActionTypes.SetupBiometricLogin,
});

export const biometricLoginSetup = (isSetup = true) => ({
  type: AuthActionTypes.BiometricLoginSetup,
  payload: { isSetup },
});
