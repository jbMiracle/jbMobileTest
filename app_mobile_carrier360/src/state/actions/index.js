import * as App from './app';
import * as Auth from './auth';
import * as UserAccess from './userAccess/UserAccessActions';

export {
  App,
  Auth,
  UserAccess,
};
