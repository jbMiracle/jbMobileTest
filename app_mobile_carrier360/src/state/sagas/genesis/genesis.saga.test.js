import SagaTester from 'redux-saga-tester';
import { App as AppActions } from '../../actions';
import genesisSaga from '.';

describe('GenesisSaga', () => {
  let sagaTester = null;
  beforeEach(() => {
    sagaTester = new SagaTester({ });
    sagaTester.start(genesisSaga);
  });

  test('the saga can start the app successfully', async () => {
    sagaTester.dispatch(AppActions.genesis());
    expect(sagaTester.getState()).toMatchSnapshot();
  });
});
