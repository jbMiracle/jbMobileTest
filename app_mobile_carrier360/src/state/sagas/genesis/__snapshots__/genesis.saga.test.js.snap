// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenesisSaga the saga can start the app successfully 1`] = `Object {}`;
